source/fe/fe_poly.cc

```diff
@@ -56,11 +56,11 @@ FE_Poly<TensorProductPolynomials<1>,1,2>::fill_fe_values
       if (flags & update_gradients && cell_similarity != CellSimilarity::translation)
         mapping.transform(fe_data.shape_gradients[k], data.shape_gradients[k],
                           mapping_data, mapping_covariant);
-    }
 
-  if (flags & update_hessians && cell_similarity != CellSimilarity::translation)
-    this->compute_2nd (mapping, cell, QProjector<1>::DataSetDescriptor::cell(),
-                       mapping_data, fe_data, data);
+      if (flags & update_hessians && cell_similarity != CellSimilarity::translation)
+        mapping.transform(fe_data.shape_hessians[k], data.shape_hessians[k],
+                          mapping_data, mapping_covariant_gradient);
```

Comment:
No, this can't be all that is needed in the general case (it is correct in the Cartesian/affine case, though). There should also be a term of the form `inv_jac_grad * fe_data.shape_gradients[k]`. I haven't checked all yet, though.